src/fileActions.jsx

```diff
@@ -209,3 +216,95 @@ export const RenameItemModal = ({ path, setPath, selected }) => {
         </Modal>
     );
 };
+
+export const EditPermissionsModal = ({ selected, path }) => {
+    const Dialogs = useDialogs();
+    const getPermissions = (str) => {
+        let permissions = 0;
+        if (str.slice(0, 1) === "r") { permissions += 4 }
+        if (str.slice(1, 2) === "w") { permissions += 2 }
+        if (str.slice(2, 3) === "x") { permissions += 1 }
+        return permissions;
+    };
+    const [owner, setOwner] = useState(selected.owner);
+    const [ownerAccess, setOwnerAccess] = useState(getPermissions(selected.permissions.slice(1, 4)));
+    const [group, setGroup] = useState(selected.group);
+    const [groupAccess, setGroupAccess] = useState(getPermissions(selected.permissions.slice(4, 7)));
+    const [otherAccess, setOtherAccess] = useState(getPermissions(selected.permissions.slice(7, 10)));
+    const [errorMessage, setErrorMessage] = useState(undefined);
+
+    const editPermissions = () => {
+        const options = { err: "message", superuser: "try" };
+        const command = ["chmod", ownerAccess + groupAccess + otherAccess, "/" + path.join("/") + "/" + selected.name];
+
+        cockpit.spawn(command, options)
+                .then(Dialogs.close, err => { setErrorMessage(err.message) });
+        cockpit.spawn(["chown", owner + ":" + group, "/" + path.join("/") + "/" + selected.name], options)
+                .then(Dialogs.close, err => { setErrorMessage(err.message) });
+    };
+
+    const permissions = [
+        { label: _("None"), value: "0" },
+        { label: _("Read-only"), value: "4" },
+        { label: _("Write-only"), value: "2" },
+        { label: _("Execute-only"), value: "1" },
+        { label: _("Read and write"), value: "6" },
+        { label: _("Read and execute"), value: "5" },
+        { label: _("Read, write and execute"), value: "7" },
+        { label: _("Write and execute"), value: "3" },
+    ];
+
+    return (
+        <Modal
+          position="top"
+          title={_("Edit permissions")}
+          isOpen
+          onClose={Dialogs.close}
+          footer={errorMessage === undefined &&
+          <>
+              <Button variant="primary" onClick={editPermissions}>{_("Change")}</Button>
+              <Button variant="link" onClick={Dialogs.close}>{_("Cancel")}</Button>
+          </>}
+        >
+            <Stack>
+                {errorMessage !== undefined &&
+                <InlineNotification
+                  type="danger"
+                  text={errorMessage}
+                  isInline
+                />}
+                <Form isHorizontal>
+                    <FormGroup label={_("Owner")}>
+                        <TextInput value={owner} onChange={setOwner} id="edit-permissions-owner" />
+                    </FormGroup>
+                    <FormGroup label={_("Owner access")}>
```

Comment:
FormGroups need a property called `fieldId` whose value is the if of the form field contained element, in this case here `edit-permissions-owner`